src/ripple/rpc/impl/Role.cpp

```diff
@@ -137,10 +133,23 @@ forwardedFor(http_request_type const& request)
             return {};
 
         found += forStr.size();
-        auto pos{it->value().find(';', forStr.size())};
-        if (pos != boost::string_view::npos)
-            return {found, pos + 1};
-        return {found, it->value().size() - forStr.size()};
+        std::size_t const pos ([&]()
+        {
+            std::size_t const pos{boost::string_view(
+                found, it->value().end() - found).find(';')};
+            if (pos == boost::string_view::npos)
+                return it->value().size() - forStr.size();
+            return pos;
+        }());
+
+        return *boost::beast::http::token_list(
+            boost::string_view(found, pos)).begin();
+    }
+
+    it = request.find("X-Forwarded-For");
+    if (it != request.end())
+    {
+        return *boost::beast::http::token_list(it->value()).begin();
```

Comment:
This used to be `beast::http::ext_list` and now it's `beast::http::token_list` ... was that part of the bug fix?

Reply:
@jwbusch Yes, "ext_list" returns extensions which are basically key=value pairs, each separated by comma. "token_list" returns single items (tokens), also separated by comma.

The X-Forwarded-For HTTP header contains IP addresses separated by columns. I don't know why I used ext_list originally, but it was a mistake.